Develop API endpoints for CRUD operations on dishes

Richard00923 opened the pull request.

Richard00923 committed:
Made corrections v2

Richard00923 committed:
Made corrections v2

Richard00923 committed:
Create API points

vmakar1134 commented on src/main/java/com/example/demo/services/ProductService.java:
```diff
@@ -0,0 +1,11 @@
+
+@RestController
+@RequestMapping("/api/product")
+public class ProductService  {
```
You should have all business logic in the service layer, as discussed at the meeting

Richard00923 committed:
Implement search functionality for retrieving receipts by various criteria (e.g., date, customer)
Design and create the database schema for receipts
Develop API endpoints for CRUD operations on receipts

vmakar1134 commented on src/main/java/com/example/demo/database/OrderItems.java:
```diff
@@ -6,28 +6,33 @@
-    private long orderId;
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "order_items_id")
+    private long orderItemsId;
```
why not just `id` instead of `orderItemsId`

The same is for the order table

vmakar1134 commented on src/main/java/com/example/demo/database/Receipt.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long receiptId;
```
just `id`

vmakar1134 commented on src/main/java/com/example/demo/database/Receipt.java:
```diff
@@ -0,0 +1,47 @@
+
+    private int totalAmount;
+
+    private Date dateTime;
```
1. use `java.time.LocalDate` instead of `java.util.Date`
2. think of a better name for the variable. What does this `dateTime` mean?

vmakar1134 commented on src/main/java/com/example/demo/database/Receipt.java:
```diff
@@ -0,0 +1,47 @@
+    public Receipt(long customerId, List<OrderItems> orderItems) {
+        this.customerId = customerId;
+        this.orderItems = orderItems;
+        calculateTotalAmount();
```
The total amount will be calculated only when we use this constructor. 
So if we use no-args constructor and setters, we won't have the total amount calculated.
That's why `calculateTotalAmount()` belongs to the `ReceiptService`

vmakar1134 commented on src/main/java/com/example/demo/services/ProductService.java:
```diff
@@ -0,0 +1,55 @@
+            return ResponseEntity.ok(products);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
```
It's better to build `ResponseEntity` in the controller
Just return `List<Product>`

vmakar1134 commented on src/main/java/com/example/demo/services/ProductService.java:
```diff
@@ -0,0 +1,55 @@
+            Product updated = productRepository.save(product);
+            return ResponseEntity.ok(updated);
+        } else {
+            return ResponseEntity.notFound().build();
```
The same here. Just return `Product` or throw some exception if none found

vmakar1134 replied:
Still not resolved

vmakar1134 commented on src/main/java/com/example/demo/services/ReceiptService.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public Receipt createReceipt(Receipt receipt) {
+        // Логіка створення чеку
```
1. There is no reason to put comments here.
2. At least comments should be in English

vmakar1134 commented on src/main/java/com/example/demo/services/ReceiptService.java:
```diff
@@ -0,0 +1,59 @@
+        if (receiptRepository.existsById(id)) {
+            receiptRepository.deleteById(id);
+        } else {
+            throw new ChangeSetPersister.NotFoundException();
```
Some weird exception. Just throw `EntityNotFoundException`

Richard00923 committed:
Implement search functionality for retrieving receipts by various criteria (e.g., date, customer)
Design and create the database schema for receipts
Develop API endpoints for CRUD operations on receipts

Richard00923 committed:
Implement search functionality for retrieving receipts by various criteria (e.g., date, customer)
Design and create the database schema for receipts
Develop API endpoints for CRUD operations on receipts

vmakar1134 commented on src/main/java/com/example/demo/controller/ProductController.java:
```diff
@@ -24,8 +25,8 @@ public ResponseEntity<String> deleteProduct(@PathVariable Long id) {
-    public ResponseEntity<List<Product>> getAllProducts() {
-        return productService.getAllProducts();
+    public ResponseEntity<List<Product>> getAllProducts() throws ChangeSetPersister.NotFoundException {
+        return (ResponseEntity<List<Product>>) productService.getAllProducts();
```
Did you test this?
It looks like it won't work

vmakar1134 commented on src/main/java/com/example/demo/database/Order.java:
```diff
@@ -1,24 +1,23 @@
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
-    @Column
+    @Column(name = "order_id")
```
just `id`

vmakar1134 commented on src/main/java/com/example/demo/database/Order.java:
```diff
@@ -1,24 +1,23 @@
 
 @Entity
-@Setter
-@Getter
```
Why did you remove `@Gettter` and `@Setter` ?

vmakar1134 commented on src/main/java/com/example/demo/database/Receipt.java:
```diff
@@ -15,13 +15,14 @@ public class Receipt {
 
-    private Date dateTime;
+    @Column
+    private LocalDate date;
```
What date is it? Is it the date of creation, or the date of update?
You should be more specific in you naming

vmakar1134 commented on src/main/java/com/example/demo/services/ReceiptService.java:
```diff
@@ -48,12 +55,24 @@ public List<Receipt> searchReceipts(Date date, Long id) {
+        receipt.setTotalAmount(total);
+    }
+
+    public Receipt createReceipt(Receipt receipt) {
```
this method doesn't make sense

Richard00923 committed:
Implement search functionality for retrieving receipts by various criteria (e.g., date, customer)
Design and create the database schema for receipts
Develop API endpoints for CRUD operations on receipts
